src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionInfo.cs

```diff
@@ -99,6 +99,15 @@ public ICollection<string> AllConnectionTypes
             }
         }
 
+        public bool HasConnectionType(string connectionType)
+        {
+            if (connectionType == null)
+            {
+                connectionType = ConnectionType.Default;
+            }
+            return ConnectionTypeToConnectionMap.ContainsKey(connectionType);

```

Comment:
So in this case we're not doing a Get. We check if it has the key. If so, we later on do a disconnect (which removes the key) and replace this ConnectionInfo with a new one, thus discarding all results. I'm struggling to see a way to do that without a method like this. Sometimes, you need to check contents of the object.